src/clib/lib/job_queue/lsf_driver.cpp

```diff
@@ -780,45 +798,15 @@ static int lsf_driver_get_job_status_shell(void *__driver, void *__job) {
 }
 
 job_status_type lsf_driver_convert_status(int lsf_status) {
-    job_status_type job_status;
-    switch (lsf_status) {
-    case JOB_STAT_NULL:
-        job_status = JOB_QUEUE_NOT_ACTIVE;
-        break;
-    case JOB_STAT_PEND:
-        job_status = JOB_QUEUE_PENDING;
-        break;
-    case JOB_STAT_SSUSP:
-        job_status = JOB_QUEUE_RUNNING;
-        break;
-    case JOB_STAT_USUSP:
-        job_status = JOB_QUEUE_RUNNING;
-        break;
-    case JOB_STAT_PSUSP:
-        job_status = JOB_QUEUE_RUNNING;
-        break;
-    case JOB_STAT_RUN:
-        job_status = JOB_QUEUE_RUNNING;
-        break;
-    case JOB_STAT_DONE:
-        job_status = JOB_QUEUE_DONE;
-        break;
-    case JOB_STAT_EXIT:
-        job_status = JOB_QUEUE_EXIT;
-        break;
-    case JOB_STAT_UNKWN: // Have lost contact with one of the daemons.
-        job_status = JOB_QUEUE_UNKNOWN;
-        break;
-    case JOB_STAT_DONE + JOB_STAT_PDONE: // = 192. JOB_STAT_PDONE: the job had a
-        // post-execution script which completed
-        // successfully.
-        job_status = JOB_QUEUE_DONE;
-        break;
-    default:
-        job_status = JOB_QUEUE_NOT_ACTIVE;
+    job_status_type job_status = JOB_QUEUE_NOT_ACTIVE;
+
+    if (auto found_status = convert_status_map.find(lsf_status);
+        found_status != convert_status_map.end())
+        job_status = found_status->second;
+    else
         util_abort("%s: unrecognized lsf status code:%d \n", __func__,
```

Comment:
This util_abort can be made into a runtime_exception

Reply:
Replaced with `exc::runtime_error(...)`